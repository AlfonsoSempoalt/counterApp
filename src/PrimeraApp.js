import React from "react";
import PropTypes from "prop-types";

const PrimeraApp = ({saludo,subtitulo}) => {
  return (
      <>
      <h1>{saludo} c:</h1>
      <h2>{subtitulo}</h2>
      </>
  )
};

PrimeraApp.propTypes = {
    saludo:PropTypes.string.isRequired
};

PrimeraApp.defaultProps = {
    subtitulo:"Buenas noches",
}

export default PrimeraApp;
