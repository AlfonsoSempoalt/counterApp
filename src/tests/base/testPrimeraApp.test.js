import PrimeraApp from "../../PrimeraApp";
import React from "react";
import { shallow } from "enzyme";
//import {render} from '@testing-library/react'

describe("test to PrimeraApp", () => {
  /*  test('should show hola', () => {
        const saludo='hola';
        const wrapper= render(<PrimeraApp saludo={saludo}/ >)
        expect(wrapper.getByText("hola")).toBeInTheDocument();
    }) */

  test("<PrimeraApp/ > should renderize properly", () => {
    const saludo = "hoola buenas noches";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("<PrimeraApp/ > test props", () => {
    const saludo = "hoola buenas noches";
    const subtitulo = "soy un subtitulo";
    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );
    const textAppSubtitles = wrapper.find("h2").text();
    expect(textAppSubtitles).toBe(subtitulo);
  });
});
