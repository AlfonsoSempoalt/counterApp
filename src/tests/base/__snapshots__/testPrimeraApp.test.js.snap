// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test to PrimeraApp <PrimeraApp/ > should renderize properly 1`] = `
<Fragment>
  <h1>
    hoola buenas noches
     c:
  </h1>
  <h2>
    Buenas noches
  </h2>
</Fragment>
`;

exports[`test to PrimeraApp <PrimeraApp/ > test props 1`] = `
<Fragment>
  <h1>
    hoola buenas noches
     c:
  </h1>
  <h2>
    soy un subtitulo
  </h2>
</Fragment>
`;
